### ISSUE TITLE
List of issues with increasing pytorch version.

### ISSUE THREAD

**Issue by dietervdb-meteo** [ecmwf/anemoi-core#523]
This is a collection of issues that prevent some users from upgrading their pytorch version to the most recent one.

Please feel free to add your own issues. Please keep updated once issues are resolved.

* max pytorch version: 2.3
   * user: RMI
   * problem: increasing pytorch version (which is linked to rocm version) above max quoted above leads degradation of memory use [see below] as well as to divergent behaviour of training loss for examples where there is no divergence using lower pytorch/rocm version

**Comment by einrone** [comment]
Hi, Dieter, Could you elaborate more? So do you mean that you get different behavior with for example torch 2.3 and a version higher than 2.3? 

I did post a path to a new and stable container on slack: /pfs/lustrep4/scratch/project_465000527/anemoi/containers/pytorch-2.7.0-rocm-6.2.4-py-3.12.9-v2.0.sif

Could you try and see if you experience same behavior if you use the container I posted above.

**Comment by dietervdb-meteo** [comment]
Comparing the above experiment between architectures it becomes clear that this issue seems lumi/rocm specific

<img width="1299" height="722" alt="Image" src="https://github.com/user-attachments/assets/57527dd2-5725-47d1-b756-795df9ec0654" />

I.e. on atos where there is actual cuda all pytorch versions coincide, and, importantly, the also coincide with the torch2.3.0-rocm6.2.4 . So it appears that is a version safe to use until this issue has been resolved.

**Comment by icedoom888** [comment]
Just adding this here to track for meteoswiss's HPC:

- Balfrin runs with Cuda Driver 525.xx
- Therefore cannot run PyTorch 2.6 + CUDA 12.6 as it requires driver ≥535

**Comment by dietervdb-meteo** [comment]
This issue has been split into sub-issues in #555 , I will close it as not to lead to duplication. Some subissues still refer here for the details.
